### DISCUSSION TITLE
[Error Maven] mvn verify me rompe todo

### DISCUSSION THREAD

**Discussion by alex-claure-m** [dds-jv/foro#178] [Consultas de herramientas] [answered]
buenos dias , tengo un problema acerca de que no me deja usar el comando mvn verify y por que por lo que entendi.... hay un error de dependencia de versiones de maven, que maven no estaria aceptando la version nueva de checkstyles y que checkstyles no acepta versiones viejas de maven, pero el tema es que yo tengo la ulitma version de maven bajado: "3.8.1" 

el Error que muestra es el siguiente:

> D:\Cursada 2021\Diseño de Sistema\TPsYejercicios\tpMacoWins\tp-macowins-master>mvn verify
> [INFO] Scanning for projects...
> [INFO]
> [INFO] -------------------< ar.edu.utn.frba.dds:ejercicio >--------------------
> [INFO] Building ejercicio 1.0-SNAPSHOT
> [INFO] --------------------------------[ jar ]---------------------------------
> [INFO]
> [INFO] --- jacoco-maven-plugin:0.7.7.201606060606:prepare-agent (default-prepare-agent) @ ejercicio ---
> [INFO] argLine set to "-javaagent:C:\\Users\\Alex\\.m2\\repository\\org\\jacoco\\org.jacoco.agent\\0.7.7.201606060606\\org.jacoco.agent-0.7.7.201606060606-runtime
> .jar=destfile=D:\\Cursada 2021\\Diseño de Sistema\\TPsYejercicios\\tpMacoWins\\tp-macowins-master\\target\\jacoco.exec"
> [INFO]
> [INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ ejercicio ---
> [INFO] Using 'UTF-8' encoding to copy filtered resources.
> [INFO] skip non existing resourceDirectory D:\Cursada 2021\Diseño de Sistema\TPsYejercicios\tpMacoWins\tp-macowins-master\src\main\resources
> [INFO]
> [INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ ejercicio ---
> [INFO] Nothing to compile - all classes are up to date
> [INFO]
> [INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ ejercicio ---
> [INFO] Using 'UTF-8' encoding to copy filtered resources.
> [INFO] skip non existing resourceDirectory D:\Cursada 2021\Diseño de Sistema\TPsYejercicios\tpMacoWins\tp-macowins-master\src\test\resources
> [INFO]
> [INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ ejercicio ---
> [INFO] Nothing to compile - all classes are up to date
> [INFO]
> [INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ ejercicio ---
> [INFO] Surefire report directory: D:\Cursada 2021\Diseño de Sistema\TPsYejercicios\tpMacoWins\tp-macowins-master\target\surefire-reports

> -------------------------------------------------------
>  T E S T S
> -------------------------------------------------------
> Running ar.edu.utn.frba.dds.macowins.PrendaTest
> Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.267 sec
> 
> Results :
> 
> Tests run: 5, Failures: 0, Errors: 0, Skipped: 0
> 
> [INFO]
> [INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ ejercicio ---
> [INFO] Building jar: D:\Cursada 2021\Diseño de Sistema\TPsYejercicios\tpMacoWins\tp-macowins-master\target\ejercicio-1.0-SNAPSHOT.jar
> [INFO]
> [INFO] --- maven-checkstyle-plugin:3.1.2:check (Validar Formateo) @ ejercicio ---
> [WARNING] Old version of checkstyle detected. Consider updating to >= v8.30
> [WARNING] For more information see: https://maven.apache.org/plugins/maven-checkstyle-plugin/examples/upgrading-checkstyle.html
> [INFO] You have 0 Checkstyle violations.
> [INFO]
> [INFO] >>> spotbugs-maven-plugin:4.2.0:check (Validar Code Smells) > :spotbugs @ ejercicio >>>
> [INFO]
> [INFO] --- spotbugs-maven-plugin:4.2.0:spotbugs (spotbugs) @ ejercicio ---
> [INFO] Fork Value is true
> [INFO] Done SpotBugs Analysis....
> [INFO]
> [INFO] <<< spotbugs-maven-plugin:4.2.0:check (Validar Code Smells) < :spotbugs @ ejercicio <<<
> [INFO]
> [INFO]
> [INFO] --- spotbugs-maven-plugin:4.2.0:check (Validar Code Smells) @ ejercicio ---
> [INFO] BugInstance size is 8
> [INFO] Error size is 0
> [INFO] Total bugs: 8
> [ERROR] Medium: Read of unwritten field ventasDelDia in ar.edu.utn.frba.dds.macowins.Negocio.ventasDelDia(Date) [ar.edu.utn.frba.dds.macowins.Negocio] At Negocio.
> java:[line 14] NP_UNWRITTEN_FIELD
> [ERROR] Medium: Unwritten field: ar.edu.utn.frba.dds.macowins.Negocio.ventasDelDia [ar.edu.utn.frba.dds.macowins.Negocio] At Negocio.java:[line 14] UWF_UNWRITTEN_
> FIELD
> [ERROR] Medium: Unused public or protected field: ar.edu.utn.frba.dds.macowins.PrendaFolder.Camisa.tipo [ar.edu.utn.frba.dds.macowins.PrendaFolder.Camisa] In Cami
> sa.java UUF_UNUSED_PUBLIC_OR_PROTECTED_FIELD
> [ERROR] Medium: Read of unwritten field unaPrenda in ar.edu.utn.frba.dds.macowins.Venta.ganancia() [ar.edu.utn.frba.dds.macowins.Venta] At Venta.java:[line 24] NP
> _UNWRITTEN_FIELD
> [ERROR] Medium: Read of unwritten field unaPrenda in ar.edu.utn.frba.dds.macowins.Venta.gananciaTotal() [ar.edu.utn.frba.dds.macowins.Venta] At Venta.java:[line 3
> 0] NP_UNWRITTEN_FIELD
> [ERROR] Medium: Read of unwritten public or protected field tipoPago in ar.edu.utn.frba.dds.macowins.Venta.recargoPorPago() [ar.edu.utn.frba.dds.macowins.Venta] A
> t Venta.java:[line 27] NP_UNWRITTEN_PUBLIC_OR_PROTECTED_FIELD
> [ERROR] Medium: Unwritten field: ar.edu.utn.frba.dds.macowins.Venta.unaPrenda [ar.edu.utn.frba.dds.macowins.Venta] At Venta.java:[line 24] UWF_UNWRITTEN_FIELD
> [ERROR] Medium: Unwritten public or protected field: ar.edu.utn.frba.dds.macowins.Venta.tipoPago [ar.edu.utn.frba.dds.macowins.Venta] At Venta.java:[line 27] UWF_
> UNWRITTEN_PUBLIC_OR_PROTECTED_FIELD
> [INFO]
> 
> 
> To see bug detail using the Spotbugs GUI, use the following command "mvn spotbugs:gui"
> 
> 
> 
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD FAILURE
> [INFO] ------------------------------------------------------------------------
> [INFO] Total time:  24.153 s
> [INFO] Finished at: 2021-04-17T09:35:04-03:00
> [INFO] ------------------------------------------------------------------------
> [ERROR] Failed to execute goal com.github.spotbugs:spotbugs-maven-plugin:4.2.0:check (Validar Code Smells) on project ejercicio: failed with 8 bugs and 0 errors 
> -> [Help 1]
> [ERROR]
> [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
> [ERROR] Re-run Maven using the -X switch to enable full debug logging.
> [ERROR]
> [ERROR] For more information about the errors and possible solutions, please read the following articles:
> [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
> 

PD: el mvn test funciona correctamente

y el checkstyle lo baje directamente de la IDE , lo cual amerita que tendria la ultima version
(imagen de la ultima version de checkstyles]
[https://ibb.co/N9M2f2R](url)
tambien hice los pasos de ir a tools, **google checks** activarlo y la de ir a code styles y elegir **unic and mac os**
*checkstyles*
[https://ibb.co/rQ2Kz2H](url)

*code styles *
[https://ibb.co/1JWpv26](url)

y la ultima version de jdk que tengo instalada es la 1.8.0.40
[https://ibb.co/KWqvfn2](url)

la verdad que adjunte todas las imagenes por que bueno ...no se jajaja
espero que me puedan ayudar a encontrar la solucion 
gracias y saludos!

**Comment by gastonprieto** [comment, marked as answer]
Buenas @alexclaureM 

Ante todo genial que mandaste todo detallado.

Fijate que en los logs se ponen unos prefijos INFO, WARNING, ERROR

Por lo que veo lo de la version entra de los que son WARNING (advertencias) que no llegna a ser errores aún, sino cosas a mejorar.

En cambio si tenes varios que dicen ERROR

```
[ERROR] Medium: Read of unwritten field ventasDelDia in ar.edu.utn.frba.dds.macowins.Negocio.ventasDelDia(Date) [ar.edu.utn.frba.dds.macowins.Negocio] At Negocio.
```

Qué son cosas que detecto spotbugs como errores. También mas abajo te comenta que podes hacer para este problema `mvn spotbugs:gui` para que te abra una aplicación de escritorio que te muestre los errores

Al finalizar Maven te dice que parte de las cosas a ejecutar fallo.

```
[ERROR] Failed to execute goal com.github.spotbugs:spotbugs-maven-plugin:4.2.0:check (Validar Code Smells) on project ejercicio: failed with 8 bugs and 0 errors
```

Fijate si con esto podes encontrar las cosas que te estan fallando

**Reply by gastonprieto** [reply]
@alexclaureM te sirvio la respuesta? si esta ok, fijate de marcarlo como respondido con la tilde.

**Reply by gastonprieto** [reply]
@alexclaureM marco como respondida, y cualquier cosa, volve a preguntarnos
